ui/pages/asset/components/token-buttons.tsx

```diff
@@ -95,8 +93,10 @@ const TokenButtons = ({
 
   const account = useSelector(getSelectedInternalAccount, isEqual);
 
-  const { chainId: multichainChainId } =
-    useMultichainSelector(getMultichainNetwork);
+  // This is only returning solana mainnet, not solana devnet?
+  const { chainId: multichainChainId } = useSelector(
+    getSelectedMultichainNetworkConfiguration,
+  );
```

Comment:
Should I create a separate fix for the original selector?
As mentioned in the PR description, it uses a hack that doesn't work well with non-evm dev/testnets.
https://github.com/MetaMask/metamask-extension/blob/2d96e51c13844590d80e0fb422b7f1828dc4f0a5/ui/selectors/multichain.ts#L206-L216

Reply:
Hmmm unfortunately this isn't easy to fix the types do not match and the `MultichainNetworkController` doesn't seem to have some fields that this selector requires (ticker, isAddressCompatible, decimals, etc)